language/tools/move-cli/src/sandbox/commands/publish.rs

```diff
@@ -3,118 +3,90 @@
 
 use crate::{
     sandbox::utils::{
-        explain_publish_changeset, explain_publish_error, get_gas_status,
+        explain_publish_changeset, explain_publish_error, get_gas_status, module,
         on_disk_state_view::OnDiskStateView,
     },
     NativeFunctionRecord,
 };
-use move_lang::{
-    self, compiled_unit::AnnotatedCompiledUnit, shared::NumericalAddress, Compiler, Flags,
-};
-use move_vm_runtime::move_vm::MoveVM;
-
 use anyhow::{bail, Result};
+use move_package::compilation::compiled_package::CompiledPackage;
+use move_vm_runtime::move_vm::MoveVM;
 use std::collections::BTreeMap;
 
 pub fn publish(
     natives: impl IntoIterator<Item = NativeFunctionRecord>,
     state: &OnDiskStateView,
-    files: &[String],
-    republish: bool,
+    package: &CompiledPackage,
```

Comment:
I'm not sure if a `CompiledPackage` has the `examples/` code in it, but if it does, a flag to `publish` to allow publishing the examples as well as the actual package code could be useful. Without that, there is (I think) no way to play with the `examples/` in the sandbox anymore.

Reply:
To be a bit more concrete: I'm trying `move package -d build; move sandbox publish` on a project with `examples/`. I see the example bytecodes in the `build/` dir, but they don't get published (which seems like a very reasonable default, but I'd like to be able to override it).

Reply:
Let me take a look at this. I think this _might_ be solved with the removal of the `package` subcommand. Once that is gone, you can pass build options to any Move CLI command, so if you wanted to publish the examples code you could do that with `move -d sandbox publish` since this is saying "compile the package in dev mode, and then publish the resulting modules." The issue today is that there's now way of doing this easily. 


I think for now, I might keep the package subcommand, but move the `BuildConfig` to the `Move` args, this way you can pass it with the `sandbox` subcommand as well.

Reply:
Should be updated now! The `BuildConfig` is now flattened into the `Move` structopt. So if you do `move -d sandbox publish` this will publish the modules in the `examples` directory.